Support different shells

dzvon opened the pull request:
As a fish shell user, I've found that our project currently only supports the bash style (`export X=Y`) of setting environment variables when I use the `-env` option. The purpose of this PR is to support other common shells.

How to determine which type of shell to use for exporting commands that set environment variables?

1. You can use the `-shell` option to specify the shell type you want explicitly.
2. If `-shell` is not provided, then use the current user's default shell type, which is we can read from the SHELL environment variable.

I _don't_ think you'll need provide the `-shell` option in most cases, unless you're manually entering a shell different from the default, which is rare, and this is the only reason the `-shell` option exist.

Any thoughts? @theplant/sre @bodhi Thank you! :)

dzvon committed:
Add support for exporting AWS credentials in different shell types

dzvon committed:
Add `-shell` flag to specify the shell type

bodhi reviewed (approved):
Good stuff, thanks @dzvon!

idealhack commented on main.go:
```diff
@@ -48,9 +50,38 @@ func arnFilename(arn string) string {
+			fmt.Printf("setenv AWS_SESSION_TOKEN %s\n", *result.Credentials.SessionToken)
+		default:
+			// For other shells, return an error
+			return errors.Errorf("unsupported shell: %s", current_shell)
```
This is changing the default behavior. It probably won't affect many people, though. I'll leave it to you.

dzvon replied:
Good point. I'll change this here for consistency with the previous behavior.

idealhack reviewed (commented):
Thanks for adding support for other shells!

dzvon committed:
Keep consistent with the previous behavior

Don't fail if the shell is not recognized, just use the export command
keeping the previous behavior.
